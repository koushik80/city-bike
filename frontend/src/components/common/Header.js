import { Container, Grid } from '@mui/material';
import React from 'react';
import headerImg from './../../image/c2.gif';
import './Header.scss';

const Header = () => {
    return (
        <header className='header'>
            <Container>
                <Grid container spacing={3}>
                    <Grid item xs={12} md={6} lg={6} >
                        <div className="header--content">
                            <h1>Helsinki City-Bike</h1>
                            <p>Welcome to use city bikes! City bikes are available in Helsinki and Espoo from the beginning of April to the end of October. Enjoy your ride!</p>
                        </div>
                    </Grid>
                    <Grid item xs={12} md={6} lg={6}>
                        <div className="header--image">
                            <img src={headerImg} alt="" />
                        </div>
                    </Grid>
                </Grid>
            </Container>
        </header>
    );
};

export default Header;