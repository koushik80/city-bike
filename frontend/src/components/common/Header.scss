.header {

    &--image {
        width: 100%;
    }
}

.header--content {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    //margin-top: 0;
    //margin-bottom: 0;
    justify-content: center;
}

h1 {
    color: #C73866;
}

.header--image {
    img {
        width: 100%;
        border-radius: 50%;
}
}