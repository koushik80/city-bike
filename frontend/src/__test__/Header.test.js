import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/common/Header';


describe("Home Header", () => {

    it("should render correctly", () => {
        const wrapper = renderer.create(<Header />);

        expect(wrapper).toMatchSnapshot();
    })
});