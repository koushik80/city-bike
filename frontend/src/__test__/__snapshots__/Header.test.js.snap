// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Header should render correctly 1`] = `
<header
  className="header"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          className="header--content"
        >
          <h1>
            Helsinki City-Bike
          </h1>
          <p>
            Welcome to use city bikes! City bikes are available in Helsinki and Espoo from the beginning of April to the end of October. Enjoy your ride!
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          className="header--image"
        >
          <img
            alt=""
            src="c2.gif"
          />
        </div>
      </div>
    </div>
  </div>
</header>
`;
